app/utils/helpers.ts

```diff
@@ -162,3 +162,45 @@ export function isMainActivity() {
         android: ShareModule?.getCurrentActivityName() === 'MainActivity',
     });
 }
+
+/**
+ * Extracts the domain name or IP address from a serverUrl.
+ *
+ * @param {string} input - The serverUrl, potentially with a port and protocol.
+ * @returns {string} The extracted domain name or IP address,
+ * or empty if no match is found.
+ *
+ * // Output:
+ * // Input: https://example-something.com => Result: example-something
+ * // Input: http://192.168.1.1 => Result: 192.168.1.1
+ * // Input: https://127.0.0.1:3000 => Result: 127.0.0.1
+ */
+export function extractDomain(input: string) {
+    const regex = /^(https?:\/\/)?([\w.-]+)(:\d+)?$/;
```

Comment:
This could cause problems if you have to servers under the same domain but using subpaths